azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -230,6 +237,11 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+      // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
+      // if the directory is not permissioned correctly already (should happen once per execution)
+      if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
```

Comment:
With this change, the default has changed. Before, users could always write to their directory. Now - due to the change to execute-as-user.c - they won't be able to until we allow them to.

The clean up change is a different problem that should be taken care of in 3.32.1 because of #1292 :+1: